File: SpringTILCode/src/main/java/com/example/springtilcode/domain/post/service/PostService.java
package com.example.springtilcode.domain.post.service;

import com.example.springtilcode.domain.post.controller.dto.response.PostResponse;
import com.example.springtilcode.domain.post.entity.Post;
import com.example.springtilcode.domain.post.exception.PostNotFoundException;
import com.example.springtilcode.domain.post.repository.PostRepository;
import com.example.springtilcode.domain.user.entity.User;
import com.example.springtilcode.domain.user.exception.UserNotFoundException;
import com.example.springtilcode.domain.user.repository.UserRepository;
import com.example.springtilcode.global.utils.SecurityUtil;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;

@Service
@RequiredArgsConstructor
public class PostService {
    private final PostRepository postRepository;
    private final SecurityUtil securityUtil;
    private final UserRepository userRepository;

    @Transactional
    public void save(String title, String content){
        User user = userRepository.findByEmail(SecurityUtil.getEmail()).orElseThrow(UserNotFoundException::new);
        Post post = Post.builder()
                .title(title)
                .content(content)
                .user(user)
                .build();
        postRepository.save(post);
    }

    @Transactional
    public void delete(Integer id){
        postRepository.deleteById(id);
    }

    @Transactional
    public PostResponse getPost(Integer id){
        Post post = postRepository.findById(id).orElseThrow(PostNotFoundException::new);
        return PostResponse.builder()
                .title(post.getTitle())
                .content(post.getContent())
                .user(post.getUser())
                .build();
    }
}

File: kotlin/src/main/kotlin/com/example/demo/domain/feed/service/PostService.kt
package com.example.demo.domain.feed.service

import com.example.demo.domain.feed.controller.dto.request.PostRequest
import com.example.demo.domain.feed.controller.dto.response.PostResponse
import com.example.demo.domain.feed.entity.Post
import com.example.demo.domain.feed.exception.PostNotFoundException
import com.example.demo.domain.feed.repository.PostRepository
import com.example.demo.domain.user.entity.User
import com.example.demo.domain.user.exception.EmailNotFoundException
import com.example.demo.domain.user.repository.UserRepository
import com.example.demo.global.error.GlobalErrorCode
import com.example.demo.global.util.SecurityUtil
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class PostService (
    private val postRepository: PostRepository,
    private val userRepository: UserRepository,
    private val securityUtil: SecurityUtil
){
    @Transactional
    fun create(request: PostRequest): PostResponse{
        val user: User = userRepository.findByEmail(securityUtil.getEmail()) ?: throw EmailNotFoundException(GlobalErrorCode.EMAIL_NOT_FOUND)
        val post: Post = postRepository.save(
            Post(
               title = request.title,
               content = request.content,
               user = user
            )
        )
        return PostResponse(post)
    }

    @Transactional
    fun delete(id: Long){
        postRepository.deleteById(id)
    }

    @Transactional
    fun getFeed(id: Long): PostResponse{
        val post: Post = postRepository.findById(id).orElseThrow{PostNotFoundException(GlobalErrorCode.BAD_REQUEST)}
        return PostResponse(post)
    }
}